apis/nucleus/src/components/listbox/listbox-selections.js

```diff
@@ -1,7 +1,46 @@
-// eslint-disable-next-line import/prefer-default-export
+const SELECTED_STATES = ['S', 'XS'];
+
+const flattenArr = (arr) => arr.reduce((prev, cur) => prev.concat(cur));
+
+function isStateSelected(qState) {
+  return SELECTED_STATES.includes(qState);
+}
+
+export function getUniques(arr) {
+  return Array.isArray(arr) ? Array.from(new Set(arr)) : undefined;
+}
+
+export function getSelectedValues(pages) {
+  if (!pages) {
+    return [];
+  }
+  const elementNbrs = pages.map((page) => {
+    const elementNumbers = page.qMatrix.map((p) => {
+      const [p0] = p;
+      return isStateSelected(p0.qState) ? p0.qElemNumber : false;
+    });
+    return elementNumbers.filter((n) => n !== false);
+  });
+  return flattenArr(elementNbrs);
+}
+
+export function applySelectionsOnPages(pages, elmNumbers, toggle = false) {
+  const getNewSelectionState = (qState) => (toggle && elmNumbers.length === 1 && isStateSelected(qState) ? 'A' : 'S');
```

Comment:
why is elmNumbers.length === 1 relevant?

I think there is a lot of improvment possible on guessing what the new state will be.
but that is also true for the current listbox and probably out of scope for this PR.

Reply:
We only want to toggle selected values on "click". For range selection (>1 values) we don't want to toggle selected values.